ci-operator/step-registry/upi/install/aws/cluster2/upi-install-aws-cluster2-commands.sh

```diff
@@ -0,0 +1,603 @@
+wait "$!"
+
+
+# ------------------------------------------------------------------------------------
+# Compute nodes
+# ------------------------------------------------------------------------------------
+
+worker_cf_template="06_cluster_worker_node.yaml"
+declare -a COMPUTE_IPS
+for index in 0 1 2; do
+    echo "Launch Additional Worker Node ${index}....."
+    worker_stack_name="${CLUSTER_NAME}-compute-${index}"
+    worker_cert_content=$(cat "${install_dir}/worker.ign" | jq -r '.ignition.security.tls.certificateAuthorities[].source')
+
+    worker_params="/tmp/aws_worker_${index}_params.json"
+    worker_output="/tmp/aws_worker_${index}_output.json"
+
+    SUBNET="PRIVATE_SUBNET_${index}"
+
+    aws_add_param_to_json "Index" "${index}" "$worker_params"
+    aws_add_param_to_json "InfrastructureName" "${INFRA_ID}" "$worker_params"
+    aws_add_param_to_json "IgnitionLocation" "https://${API_SERVER_DNS_NAME}:22623/config/worker" "$worker_params"
+    aws_add_param_to_json "Subnet" "${!SUBNET}" "$worker_params"
+    aws_add_param_to_json "CertificateAuthorities" "${worker_cert_content}" "$worker_params"
+    aws_add_param_to_json "WorkerSecurityGroupId" "${WORKER_SECURITY_GROUP_ID}" "$worker_params"
+    aws_add_param_to_json "WorkerInstanceProfileName" "${WORKER_INSTANCE_PROFILE}" "$worker_params"
+    aws_add_param_to_json "RhcosAmi" "${RHCOS_AMI_WORKER}" "$worker_params"
+    aws_add_param_to_json "WorkerInstanceType" "${WORKER_INSTANCE_TYPE}" "$worker_params"
+    extra_options=" --capabilities CAPABILITY_NAMED_IAM --tags \"${TAGS}\" "
+
+    cat "$worker_params"
+    echo "${worker_stack_name}" >> "${SHARED_DIR}/to_be_removed_cf_list_stage_1"
+    aws_create_stack ${AWS_REGION} ${worker_stack_name} \
+        "file://${upi_aws_cloudformation_template_dir}/${worker_cf_template}" \
+        "file://${worker_params}" \
+        "${extra_options}" \
+        "${worker_output}" || exit 2
+
+    COMPUTE_IP=$(cat ${worker_output} | jq -r '.Stacks[].Outputs[] | select(.OutputKey=="PrivateIP") | .OutputValue')
+    COMPUTE_IPS+=("$COMPUTE_IP")
+done
+
+
+echo "bootstrap: ${BOOTSTRAP_IP} control-plane: ${CONTROL_PLANE_0_IP} ${CONTROL_PLANE_1_IP} ${CONTROL_PLANE_2_IP} compute: ${COMPUTE_IPS[0]} ${COMPUTE_IPS[1]} ${COMPUTE_IPS[2]}"
+
+# ------------------------------------------------------------------------------------
+# Approve CSRs
+# ------------------------------------------------------------------------------------
+
+export KUBECONFIG="${install_dir}/auth/kubeconfig"
+
+
+#Actaully if bootkube is still running, all nodes's csr would be signed automatically, but once bootkube is done, would not approve any csr
+#Check `approve-csr` service on bootstrap - /usr/local/bin/approve-csr.sh
+#So have to run some wait and approve here
+wait_and_approve "master" "3" || exit 2
+wait_and_approve "worker" "3" || exit 2
+
+
+cloudfront_base_url=$(head -n 1 /var/run/vault/image-registry/cloudfront-base-url)
```

Comment:
Is it possible to split the cloudfront configuration part into a separate ref step

Reply:
Sure, but we still need to config an empty dir for Image Registry https://docs.openshift.com/container-platform/4.10/installing/installing_aws/installing-aws-user-infra.html#installation-registry-storage-non-production_installing-aws-user-infra

Reply:
I prefer to keep the install process as simple as possible, allow workflow developer to opt-in image-registry storage configuration as desire, also avoid binding vault with the upi ref step.

Reply:
updated, patched an empty dir.